### DISCUSSION TITLE
Données restreintes 🔐

### DISCUSSION THREAD

**Discussion by agarrone** [datagouv/data.gouv.fr#1198] [Ideas]
Plusieurs équipes ont fait remonter le besoin de référencer des données restreintes sur data.gouv.fr. 
Cette discussion vise à permettre à chaque équipe de clarifier et spécifier son besoin.

**Comment by Haelle** [comment]
### Côté Pôle API - DINUM

Vous connaissez déjà le sujet, mais refaisons une synthèse ici : 

**L'objectif final est la découvrabilité de la donnée en général.** 

Pour cela nous avons aujourd'hui le site api.gouv.fr qui a été initialement conçu pour adresser les données par API (vs fichier sur data.gouv.fr), mais aujourd'hui il complète les manque que peux avoir data.gouv.fr aussi sur les données restreintes (par API uniquement).

- découvrabilité des données en accès restreint (par API et par fichier bientôt pour la pro-activité côté Pôle API) ;
- avoir un seul point de recherche national sur la donnée (ouverte/fermée et API/fichier) ;
- intégrer un process d'habilitation aux données retreintes au niveau de complexité produit de Datapass (qui est en train d'évoluer pour habiliter aussi l'accès aux fichiers en accès restreints) ;
- _secondaire:_ gestion des tokens commune avec le Pôle API pour potentiellement avoir une réutilisation pour avoir des quota privilégié aux futures API data.gouv.fr aux administrations

**Comment by agarrone** [comment]
Hello, 

Voici l'état de réflexions sur les API et jeux de données restreints sur data.gouv.fr 
[Vidéo d'explication](https://www.loom.com/share/49ce9e4278f842d19ffd86c2d3a20e04?sid=9664c76a-c6fe-43b1-8650-0cc85479f7a5)
[Maquettes](https://www.figma.com/file/9Ai2c1JrRyFNKDSJOXN9wM/%F0%9F%92%BB-Api-et-donn%C3%A9es-restreintes?type=design&node-id=117%3A12719&mode=design&t=1rwZf55nlqaR9IQf-1)
Vos retours nous serions précieux 🙏 
Notamment @abulte @Haelle @AntoineAugusti @XavierJp @cquest @streino et bien d'autres :)

**Reply by cquest** [reply]
Vraiment, mais vraiment, pas chaud à faire figurer sur data.gouv des données non ouvertes.
Qui sont ces "plusieurs équipes" ?
Qui décide pour ce genre de changement majeur ?

**Reply by AntoineAugusti** [reply]
Merci pour le partage et d'avoir pris le temps de faire une rapide vidéo de présentation.

## API

- Pour un JDD avec une API associée, l'utilisateur risque d'être confus entre l'API automatique proposée par data.gouv.fr et l'API officielle (déjà constaté sur transport.data.gouv.fr avec des conversions automatiques).
- L'onglet "API" indique-t-il uniquement l'API officielle exposant les données du JDD et non une réutilisation de cette API ?
- Comment gérer la recherche fulltext quand tu as un JDD et une API (par exemple chercher "SIREN") ?
- Il semble y avoir des discussions sur la fiche JDD et sur la fiche API, il me semble que certaines discussions vont être communes aux 2 (incohérence des données, demande de modification du service etc) mais vont se retrouver uniquement d'un côté.

## Données restreintes

La présentation montre qu'il est possible de présenter des données en cours de publication/disponibles sous habilitation. On expose ainsi à tous le titre, le producteur, une description et quelques métadonnées.

Est-il prévu de répondre aux cas d'usages suivants :
- avoir des données disponibles à certains partenaires (par exemple entre plusieurs utilisateurs) mais non découvrables par le public (pas disponible dans la recherche, pas possible de demander accès)
- avoir un JDD avec un mix de données ouvertes à tous et certaines ressources restreintes
- avoir des réutilisations ou discussions partagées uniquement au producteur (partage d'informations ou de données privilégiées)

**Reply by XavierJp** [reply]
Alors bravo, je trouve ca super chouette et ta video est très bien faite @agarrone !

Par contre je suis d'accord avec @AntoineAugusti la distinction API "externe" vs API "automatique" est complexe a comprendre. La distinction vient de nos mécaniques internes et cela ne devrait pas avoir d'impact sur l’UX.

Finalement tu as divers moyens d'accéder a un JDD (téléchargement, visualisation in-site, et API). Les API sont un moyen parmi d'autres d'accèder à un jdd, et l'API datagouv est une API parmi d'autres. 

Il va falloir aider les usagers a choisir entre chaque API. Exemple : l'API datagouv est top pour récupérer des lignes, mais ce n'est pas forcément l'API a utiliser pour construire un moteur de recherche fulltext etc.

**Reply by abulte** [reply]
Merci @agarrone !

Re données accès restreintes et Ecosphères (sans concertation avec équipe à date), je pense évidemment au mapping avec INSPIRE.

Les métadonnées concernées sont "Conditions applicable à l’accès et l’utilisation" et "Restriction d’accès public" (cf IX. Contraintes en matière d’accès et d’utilisation du [guide métadonnées CNIG](https://cnig.gouv.fr/IMG/pdf/guide-de-saisie-des-elements-de-metadonnees-inspire-v2.0-allege.pdf)).

Dans ce document, je trouve l'exemple au IX.3.4. "Cas des données présentant une autre restriction d’accès public" re les relevés des traces des ours des Pyrénées assez parlant.

Il faudra certainement prévoir un champ (semi ?) libre si on veut faire un mapping INSPIRE qui explique à l'usager pourquoi une donnée n'est pas ouverte, en plus d'un booléen restreint oui/non ; là où il y a "Données personnelles" ou "Document non achevé" dans les maquettes.

Je vais essayer de voir si on peut récupérer une liste de valeurs possibles sur le stock Ecosphères existant.

Pour référence sur cette métadonnée https://github.com/ecolabdata/ecospheres/issues/51

Re les API, je me demande si on peut mapper ça à la notion de "services de données géographiques" (cf "II.3. Type de la ressource" du guide et [analyse inspire vs data.gouv.fr](https://github.com/ecolabdata/ecospheres/blob/8c6396ee56e6c10f1394f8aec851718fc792fc4d/doc/inspire_vs_data_gouv.md#type-de-la-ressource)), mais à date je ne crois pas qu'on ait beaucoup creusé cette métadonnée.

**Reply by agarrone** [reply]
Hello, merci pour vos retours ! 

On a depuis pivoté vers l'approche suivante : 

On considère les APIS comme des réutilisations. 
C'est d'ailleurs le cas aujourd'hui il existe un type de réutilisations "API" ([exemple](https://www.data.gouv.fr/fr/reuses/api-donnees-foncieres/)).

En back on stockerai ça comme des objets à part pour gérer toutes les métadonnées spécifiques mais pour l'utilisateur il s'agirait d'une application construite sur la donnée comme peut l'être une cartographie par exemple. 

Les APIs automatiques seraient traitées de façon complètement différentes comme un moyen de consommer la données probablement dans explore : les données tabulaires sont prévisualisables et requetables etc. 

[Voici les maquettes](https://www.figma.com/file/9Ai2c1JrRyFNKDSJOXN9wM/%F0%9F%92%BB-Api-et-donn%C3%A9es-restreintes?type=design&node-id=284%3A12253&mode=design&t=6maWcev3QtaUQ9Ut-1)

En espérant que ça vous semble cohérent aussi :)

**Reply by streino** [reply]
Merci @agarrone pour le partage.

Pour les données restreintes, @abulte a soulevé le point le plus important pour Ecosphères, à savoir l'utilisation des métadonnées existantes pour déterminer les conditions d'accès.

Pour les APIs, comment allez-vous gérer le cas d'une API sans JDD existant à associer ?

**Reply by abulte** [reply]
> Je vais essayer de voir si on peut récupérer une liste de valeurs possibles sur le stock Ecosphères existant.

Voir ici. Je ne connais pas la taille du stock sur lequel je travaille vs la réalité, mais pour l'instant seulement 6 JDD seraient concernés 🔐 : https://github.com/ecolabdata/ecospheres/issues/51#issuecomment-1954347903

**Reply by agarrone** [reply]
> Vraiment, mais vraiment, pas chaud à faire figurer sur data.gouv des données non ouvertes. Qui sont ces "plusieurs équipes" ? Qui décide pour ce genre de changement majeur ?

C'est une décision de longue date maintenant, issue notamment de l'expérimentation catalogue.data.gouv.fr qui a montré le besoin et l'opportunité de référencer des données restreintes sur data.gouv.fr et sur le besoin d'avoir un catalogue commun pour jeux de données et API (fusion de data.gouv.fr et d'api.gouv.fr).

Ces décisions sont prises par l'équipe data.gouv.fr et ont déjà été annoncées à la communauté notamment lors de démos publiques et de la publication de la feuille de route.

Les personnes sollicitées ici font partie de la "communauté tech" de data.gouv.fr et travaillent sur des produits gravitant autour de data.gouv.fr (transport.data.gouv.fr, ecologie.data.gouv.fr, annuaire des entreprises, etc.).

**Reply by agarrone** [reply]
> Est-il prévu de répondre aux cas d'usages suivants :
avoir des données disponibles à certains partenaires (par exemple entre plusieurs utilisateurs) mais non découvrables par le public (pas disponible dans la recherche, pas possible de demander accès)
avoir un JDD avec un mix de données ouvertes à tous et certaines ressources restreintes
avoir des réutilisations ou discussions partagées uniquement au producteur (partage d'informations ou de données privilégiées)

Non ces cas d'usages ne sont pas prévus.

**Reply by agarrone** [reply]
> Par contre je suis d'accord avec @AntoineAugusti la distinction API "externe" vs API "automatique" est complexe a comprendre. La distinction vient de nos mécaniques internes et cela ne devrait pas avoir d'impact sur l’UX.

Complètement d'accord, en ésperant que la seconde version vous semble plus cohérente :)

**Reply by XavierJp** [reply]
Désolé @agarrone mais crois que je trouve le pivot sur les réutilisations pas évident : 
- le terme d’API comporte une notion de moyen d'accès, finalement un peu au meme niveau qu'un téléchargement de fichier. Pas sur que les utilisateurs pensent a ouvrir l'onglet réutilisation pour y chercher une API 
- comment tu gères les API qui ne sont pas liées à un fichier ou un jdd ? Comment référencer une API qui fait de l'écriture par exemple ? Ou une API comme franceConnect ou meme une API de chatbot ?

**Reply by agarrone** [reply]
> Désolé @agarrone mais crois que je trouve le pivot sur les réutilisations pas évident : 
> 

Pas de soucis merci du retour ! 

> - le terme d’API comporte une notion de moyen d'accès, finalement un peu au meme niveau qu'un téléchargement de fichier. Pas sur que les utilisateurs pensent a ouvrir l'onglet réutilisation pour y chercher une API 
> 

Je suis d'accord que c'est pas évident, on s'interroge sur le fait de renommer cet onglet. 
Tu le verrais comment toi ? 

> - comment tu gères les API qui ne sont pas liées à un fichier ou un jdd ? Comment référencer une API qui fait de l'écriture par exemple ? Ou une API comme franceConnect ou meme une API de chatbot ?

En théorie les APIs devraient être associé à un jeu de données y compris restreint à terme. 
Pour le type d'APIs que tu évoque c'est pas simple mais je me demande si ces APIs ont leur place sur data.gouv.fr.

**Reply by cquest** [reply]
Les API sont un service qui **peut** permettre d'accéder ou pas à des données.

Je pense que pour data.gouv (data) seules des API permettant d'accéder à des données (ouvertes) devraient être répertoriées pour ne pas tout mélanger.

Là aussi il faut distinguer les API qui fournissent un service en plus sur les données ou pas.

Exemple: l'API de géocodage de la BAN fait bien plus que fournir un accès aux données aux données de la BAN. C'est une forme de réutilisation des données avec une valeur ajoutée quasiment plus importante que les données.

---

Je ne suis pas chaud du tout pour le "tout API", je l'ai souvent exprimé (https://cq94.medium.com/les-api-cest-bien-en-abuser-ca-craint-b5d1c92b32f2), je vais donc me répéter.

Les API posent les problèmes suivants:
- la création de dépendance pour les ré-utilisateurs des données -> si l'API est indispo cela les impacte
- la création d'une charge bien plus importante pour le diffuseur -> il doit avoir une SLA au top pour éviter d'impacter les ré-utilisateurs
- la gestion pour le diffuseur du scaling: on tombe vite dans la fixation de quota, la création de comptes pour les gérer, ou de whitelists, etc... où alors c'est un chèque en blanc sur les besoins de ressources qui seront nécessaires et une SLA de plus en plus difficile à maintenir.

On a malheureusement pas retenu la leçon avec l'API de la BAN ou une version de démo d'un outil de géocodage (addok) est petit à petit devenue un service critique qu'il n'aurait pas dû devenir.

Les API restent pertinentes dans des cas finalement limités :
- accès contrôlé aux données
- données de très grand volume
- données avec une fréquence très élevée de mise à jour

Le diffusion avec des formats opimisés cloud (parquet, cog, flatgeobuf, pmtiles, etc) diffusés en http simple (ou s3) est une alternative intermédiaire à très fort potentiel... mais qui ne semble intéresser personne, malheureusement.

**Comment by agarrone** [comment]
Hello, 
Voici [la dernière version](https://www.figma.com/file/5RmlKC8xkB0uT1MusIvFNz/API-sur-data.gouv.fr-(shared)?type=design&node-id=0%3A1960&mode=design&t=cuNSnxfxD332wcb4-1) qui a été envoyé aux beta testeur de data.gouv.fr. 

![image](https://github.com/etalab/data.gouv.fr/assets/60264344/78ae03a9-3c27-4de2-bafb-a229456ce64b)
